themes/default/content/blog/zephyr-code-placement-and-stacks/index.md

```diff
@@ -0,0 +1,133 @@
+
+<!--more-->
+
+Here are links to all the blog posts in the series:
+
+**Zephyr: Code Placement and Stacks** (this post)
+
+Zephyr: Structuring Their Pulumi Projects
+
+Zephyr: Moving Infrastructure State Between Projects
+
+Zephyr: Tying Stacks Together with Stack References
+
+Zephyr: Evolving the Application
+
+Zephyr: Adding Pulumi Deployments
+
+Zephyr: Refactoring for Reuse
+
+Zephyr: Local Testing with Pulumi
+
+## What is Zephyr?
+
+Zephyr is short for Zephyr Archaeotech Emporium, the fictional company in our scenario. <!--Aaron, need details on the story/background here-->
+
+## The application: Zephyr's online store
+
+Zephyr's application is the second generation of their online store. Their first generation was a monolith that they manually deployed, but as they prepare for the next phase of their growth---driven by robust sales of arcane artifacts---they've moved to a containerized microservices architecture. Here's a diagram of their second-generation application architecture:
+
+![Placeholder Image](meta.png)
+
+Astute readers may note that this looks pretty similar to [this application](https://github.com/aws-containers/retail-store-sample-app/); in truth, we're using a fork of the AWS Containers retail store example as the basis for our fictional scenario (many thanks to the AWS Containers DA team for their hard work here!).
+
+## Deploying the application with Pulumi
+
+As part of the switch to their new architecture, Zephyr has decided they want to use Pulumi to manage the infrastructure and application. As they prepared to embark on using Pulumi, a couple of questions came up for the Zephyr team:
+
+* Where should they store their Pulumi code? The ability for Zephyr's developers to use a full programming language to manage the infrastructure and deployment of their online store is a huge plus, but should this code go in the same repository as their application code? Or a different repository?
+* How do they address the need for multiple environments? It's pretty clear to the Zephyr development team that they'll need multiple instances of the online store (for things like testing, QA, or development). What's the best way to handle this when using Pulumi?
+
+Let's examine each of these issues in a bit more detail, and see how Zephyr is choosing to proceed.
+
+### Storing their Pulumi code
+
+When it comes to answering the question of where to store Pulumi code relative to the application(s) it supports, there are three basic options:
+
+1. In the same repository and same branch as your application code
+2. In the same repository as your application code, but in a different branch
+3. In a separate repository from your application code
+
+Each of these options has its own advantages and disadvantages. For example, ...
```

Comment:
Consider the formatting pattern above for list items and put the pros and cons inline, e.g.

1. *In the same repository and branch as the app code:* This works well for... but doesn't work as well when...

Reply:
Addressed in 35635a95